solar_system.rb

```diff
@@ -0,0 +1,58 @@
+require 'pry'
+
+class SolarSystem
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  attr_reader :star_name, :planets
+
+  def add_planet(new_planet)
+    if new_planet.class != Planet
+      raise ArgumentError, "Not a planet"
+    else
+      @planets.push(new_planet)
+    end
+  end
+
+  def list_planets
+    result = "Planets orbiting #{@star_name}\n"
+    planet_list = @planets
+
+    planet_list.each_with_index do |item, i|
+      result << "#{i+1}. #{item.name}\n"
+    end
+
+    return result
+  end
+
+  def find_planet_by_name(name_of_planet)
+
+    result = @planets.select do |planet|
+      planet.name.downcase == name_of_planet.downcase
+    end
+
+    if result.length == 0
```

Comment:
If result is a Planet, it's not going to have a length.  You should do `if result.nil?`